packages/donor/src/screens/UpcommingDonation/SameDayDonationUtil.ts

```diff
@@ -0,0 +1,19 @@
+import { BookedAppointment, DateUtils } from "@zm-blood-components/common";
+
+export function shouldDisplaySameDayDonationPopup(
+  now: Date,
+  bookedAppointment: BookedAppointment
+) {
+  const donationStartTime = new Date(bookedAppointment.donationStartTimeMillis);
+  const bookingTime = new Date(bookedAppointment.bookingTimeMillis);
+  if (!DateUtils.isSameDay(now, donationStartTime)) {
+    return false;
+  }
+
+  // If appointment was booked more than 10s ago
+  if (now.getTime() - bookingTime.getTime() > 10_000) {
+    return false;
+  }
```

Comment:
what is that for?

Reply:
Without this, a user could schedule on Sunday an appointment starting on Monday, and then re-open the app on Monday and see the popup.

Here I verify that the appointment was booked very recently

Reply:
Added a comment

Reply:
but this code is executed only if the appointment is in the same day, so in this usecase the code will not get there at all.